packages/video_player/video_player_avfoundation/example/ios/RunnerTests/VideoPlayerTests.m

```diff
@@ -61,6 +62,63 @@ - (CGAffineTransform)preferredTransform {
 @interface VideoPlayerTests : XCTestCase
 @end
 
+@interface AVPlayerMock : AVPlayer
+@property(assign) NSMutableArray *beforeToleranceArray;
+@property(assign) NSMutableArray *afterToleranceArray;
+
+@end
+
+@implementation AVPlayerMock
+
+- (instancetype)initWithBeforeArray:(NSMutableArray *)beforeArray
+                         afterArray:(NSMutableArray *)afterArray
+                         playerItem:(AVPlayerItem *)playerItem {
+  self = [super init];
+  if (self) {
+    _beforeToleranceArray = beforeArray;
+    _afterToleranceArray = afterArray;
+  }
+
+  return self;
+}
+
+- (void)seekToTime:(CMTime)time
+      toleranceBefore:(CMTime)toleranceBefore
+       toleranceAfter:(CMTime)toleranceAfter
+    completionHandler:(void (^)(BOOL finished))completionHandler {
+  [_beforeToleranceArray addObject:[NSNumber numberWithLong:toleranceBefore.value]];
+  [_afterToleranceArray addObject:[NSNumber numberWithLong:toleranceAfter.value]];
+  completionHandler(YES);
+}
+
+@end
+
+@interface AVPlayerFactoryFake : NSObject <AVPlayerFactoryProtocol>
```

Comment:
nit: be consistent about fake vs mock usage

Reply:
this can also use ocmock

Reply:
Why would we want to use OCMock instead of a trivial fake? Mocking is generally considered to be the last resort option, not the first resort.

Reply:
> Mocking is generally considered to be the last resort option, not the first resort.

Hmmm, I wasn't aware of this guideline. The benefits I see is, OCMock saves lots of boilerplate like these 2 classes, and it's also more consistent with other tests we have. This is not mocking static methods, it won't have the drawback of side effects across different tests we mentioned earlier.